### PR TITLE
made GSD windows support optional, now raises Error

### LINKED ISSUE #1923: GSD & Windows
See #1914 

TODO: Add nice handling of GSD under Windows

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1923

Changes made in this Pull Request:
 - now optionally import GSD modules

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
@tylerjereddy I don't have a windows box, but I think this should work.. some tests will need to be marked as xfail for windows (if they're not already)

File: package/MDAnalysis/coordinates/__init__.py [hunk 1]
```diff
@@ -746,6 +747,13 @@ class can choose an appropriate reader automatically.
 from . import TXYZ
 from . import memory
 from . import MMTF
-from . import GSD
 from . import null
 from . import dummy
+
+if not os.name == 'nt':
```
Comment (tylerjereddy) [inline package/MDAnalysis/coordinates/__init__.py]: You probably want `!=` here

File: package/MDAnalysis/topology/__init__.py [hunk 2]
```diff
@@ -301,5 +303,12 @@
 from . import DLPolyParser
 from . import HoomdXMLParser
 from . import MMTFParser
-from . import GSDParser
 from . import MinimalParser
+
+if not os.name == 'nt':
```
Comment (tylerjereddy) [inline package/MDAnalysis/topology/__init__.py]: Probably want `!=` here

**Comment by richardjgowers** [comment]
Ok my last fix wasn't any good as it changed the namespace (ie `MDAnalysis.coordinates.GSD.GSDReader` wouldn't exist) so it might break scripts rather than fail gracefully as intended.  This fix is even more fixier than before.

**Comment by tylerjereddy** [comment]
Ok, in it goes. I think we can safely ignore codecov patch diff coverage being a little off target here.

This should mean we can safely remove `gsd` installs to Windows builds, which were only acting as "mock" imports really.

**Pull request merged**